diff --git a/binutils/dwarf.c b/binutils/dwarf.c
index 4f695bf2bca..bb917e9ddbd 100644
--- a/binutils/dwarf.c
+++ b/binutils/dwarf.c
@@ -4491,7 +4491,7 @@ display_debug_lines_raw (struct dwarf_section *  section,
   while (data < end)
     {
       static DWARF2_Internal_LineInfo saved_linfo;
-      DWARF2_Internal_LineInfo linfo;
+      DWARF2_Internal_LineInfo linfo = {0};
       unsigned char *standard_opcodes;
       unsigned char *end_of_sequence;
       int i;
diff --git a/binutils/readelf.c b/binutils/readelf.c
index 97d72d0b95f..ae82adf0841 100644
--- a/binutils/readelf.c
+++ b/binutils/readelf.c
@@ -368,7 +368,7 @@ enum versioned_symbol_info
 static int
 fseek64 (FILE *stream, int64_t offset, int whence)
 {
-#if defined (HAVE_FSEEKO64)
+#if 0
   off64_t o = offset;
   if (o != offset)
     {
@@ -376,7 +376,7 @@ fseek64 (FILE *stream, int64_t offset, int whence)
       return -1;
     }
   return fseeko64 (stream, o, whence);
-#elif defined (HAVE_FSEEKO)
+#elif 0
   off_t o = offset;
   if (o != offset)
     {
diff --git a/libiberty/xmalloc.c b/libiberty/xmalloc.c
index c30b8966ada..bdcf52d20fa 100644
--- a/libiberty/xmalloc.c
+++ b/libiberty/xmalloc.c
@@ -119,7 +119,7 @@ xmalloc_set_program_name (const char *s)
 void
 xmalloc_failed (size_t size)
 {
-#ifdef HAVE_SBRK
+#if 0
   size_t allocated;
 
   if (first_break != NULL)
