diff --git a/src/string/stpcpy.c b/src/string/stpcpy.c
index 4db46a9e..ca9b385f 100644
--- a/src/string/stpcpy.c
+++ b/src/string/stpcpy.c
@@ -9,18 +9,6 @@
 
 char *__stpcpy(char *restrict d, const char *restrict s)
 {
-#ifdef __GNUC__
-	typedef size_t __attribute__((__may_alias__)) word;
-	word *wd;
-	const word *ws;
-	if ((uintptr_t)s % ALIGN == (uintptr_t)d % ALIGN) {
-		for (; (uintptr_t)s % ALIGN; s++, d++)
-			if (!(*d=*s)) return d;
-		wd=(void *)d; ws=(const void *)s;
-		for (; !HASZERO(*ws); *wd++ = *ws++);
-		d=(void *)wd; s=(const void *)ws;
-	}
-#endif
 	for (; (*d=*s); s++, d++);
 
 	return d;
